### PR TITLE
Ports: Add SDL 1.2 compatibility layer (sdl12-compat)

### PR DESCRIPTION
**Pull request by lightspot21**
This is a port of https://github.com/libsdl-org/sdl12-compat for SerenityOS. It provides a way to port games linked against SDL 1.2, while actually using SDL 2.x in the background

I can't find something to test this yet, so if anyone can run this and see if it works I'd be grateful!

### CODE REVIEW

File: Ports/SDL2_compat/package.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=SDL2_compat
```
Comment (timschumi) [inline Ports/SDL2_compat/package.sh:2]: Upstream named this piece of software `sdl12-compat`, and I don't really see a reason to deviate from that.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/SDL2_compat/package.sh (hunk 1 above)
Comment (lightspot21) [reply Ports/SDL2_compat/package.sh:2]: I did it for consistency, but I can change that. Fixing.

**Comment by timschumi** [comment]
I can confirm that it works, Angband runs just fine on it (once porting a bunch of addons). However, I'm getting some `pthread` issues while loading the library, but I'm not sure whose fault that is.

**Comment by lightspot21** [comment]
@timschumi great! What kind of pthread issues?

**Comment by GMTA** [comment]
@lightspot21 you could take a look at the Tux Racer and GLtron ports, which are manually patched to support SDL2.

**Comment by timschumi** [comment]
> @timschumi great! What kind of pthread issues?

Basically, it was a basic assert failure in our pthread forwarding code, due to how that is set up.

The main idea is that our LibC has function stubs that assert if a reference has been set, and then forward the function call to that reference. Those references get filled in by LibPthread once it is loaded.

However, this usually requires us to manually link LibPthread, as self-configuring builds will just see the pthread functions in LibC and assume them to be good, and LibC doesn't automatically load LibPthread either.

This means that the solution would be to just add a `-lpthread` somewhere, but I'd like to have a more universal and less hacky solution, so I'm currently looking into whether we can remove that alltogether.

File: Ports/AvailablePorts.md [hunk 2]
```diff
@@ -198,6 +198,7 @@ This list is also available at [ports.serenityos.net](https://ports.serenityos.n
 | [`ruby`](ruby/)                                     | Ruby                                                            | 3.0.4                    | https://www.ruby-lang.org/                                                     |
 | [`sam`](sam/)                                       | Software Automatic Mouth (SAM)                                  | c86ea39                  | https://github.com/vidarh/SAM                                                  |
 | [`scummvm`](scummvm/)                               | ScummVM                                                         | 2.5.1                    | https://www.scummvm.org/                                                       |
+| [`sdl12-compat`](sdl12-compat/)                     | SDL2 compatibility layer for SDL 1.2 games                      | 1.2.52                   | https://github.com/libsdl-org/sdl12-compat/                                    |
```
Comment (timschumi) [inline Ports/AvailablePorts.md:201]: This should go right before `SDL2`, I think.

**Review by timschumi — CHANGES_REQUESTED** [review]
Last small nitpick, then this should be ready as far as I can tell.

File: Ports/AvailablePorts.md (hunk 2 above)
Comment (lightspot21) [reply Ports/AvailablePorts.md]: And done!

**Pull request merged**